platform/tests/Platform.ApiTests/Steps/EstablishmentTrustsSteps.cs

```diff
@@ -35,118 +32,139 @@ private void GivenAValidTrustRequestWithId(string id)
+            RequestUri = new Uri("/api/trusts/suggest", UriKind.Relative),
+            Method = HttpMethod.Post,
+            Content = new StringContent(content.ToJson(), Encoding.UTF8, "application/json")
         });
     }
 
-    [Then("the trusts query result should be ok")]
-    private void ThenTheTrustsQueryResultShouldBeOk()
+    [When("I submit the trust request")]
+    private async Task WhenISubmitTheTrustsRequest()
     {
-        var result = _api[QueryRequestKey].Response;
-
-        result.Should().NotBeNull();
-        result.StatusCode.Should().Be(HttpStatusCode.OK);
+        await _api.Send();
     }
 
-    [Given("a valid trusts search request")]
-    private void GivenAValidTrustsSearchRequest()
+    [Then("the trust result should be correct")]
+    private async Task ThenTheTrustResultShouldBeCorrect()
     {
-        _api.CreateRequest(SearchRequestKey, new HttpRequestMessage
-        {
-            RequestUri = new Uri("/api/trusts/search", UriKind.Relative),
-            Method = HttpMethod.Post
-        });
+        var response = _api[RequestKey].Response;
+
+        response.Should().NotBeNull();
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        var content = await response.Content.ReadAsByteArrayAsync();
+        var result = content.FromJson<Trust>();
+
+        result.CompanyNumber.Should().Be("7539918");
+        result.TrustName.Should().Be("Test Company/Trust  1");
     }
 
-    [Then("the trusts search result should be ok")]
-    private void ThenTheTrustsSearchResultShouldBeOk()
+    [Then("the trust result should be not found")]
+    private void ThenTheTrustResultShouldBeNotFound()
     {
-        var result = _api[SearchRequestKey].Response;
+        var result = _api[RequestKey].Response;
 
         result.Should().NotBeNull();
-        result.StatusCode.Should().Be(HttpStatusCode.OK);
+        result.StatusCode.Should().Be(HttpStatusCode.NotFound);
     }
 
-    [Given("an invalid trusts suggest request")]
-    private void GivenAnInvalidTrustsSuggestRequest()
+    [Then("the trust suggest result should be correct")]
+    private async Task ThenTheTrustsSuggestResultShouldBeCorrect()
     {
-        var content = new { Size = 0 };
+        var response = _api[SuggestRequestKey].Response;
 
-        _api.CreateRequest(SuggestInvalidRequestKey, new HttpRequestMessage
-        {
-            RequestUri = new Uri("/api/trusts/suggest", UriKind.Relative),
-            Method = HttpMethod.Post,
-            Content = new StringContent(content.ToJson(), Encoding.UTF8, "application/json")
-        });
+        response.Should().NotBeNull();
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        var content = await response.Content.ReadAsByteArrayAsync();
+        var results = content.FromJson<SuggestResponse<Trust>>().Results;
+        var result = results.FirstOrDefault();
+        result.Should().NotBeNull();
+
+        result?.Text.Should().Be("*7539918*");
```

Comment:
Please can the asserted values move to the feature using `table.CompareToSet()`?